src/components/profile.jsx

```diff
@@ -1,32 +1,45 @@
-const Profile = () => (
-  <div className="container">
-    <div className="row">
+import { useSelector } from 'react-redux';
+import PropTypes from 'prop-types';
 
-      <div className="col-6">
-        <h2>My Mission</h2>
-        <table className="table table-bordered">
-          <tbody>
-            <tr><th>Telstar</th></tr>
-            <tr><th>SES</th></tr>
-            <tr><th>AsiasSat</th></tr>
-            <tr><th>ABS</th></tr>
-          </tbody>
+const Profile = ({ missionprofile }) => {
+  const { rocketStore } = useSelector((store) => store.rocket);
+  return (
+    <div className="container">
+      <div className="row">
+        <div className="col-6">
+          <h2>My Mission</h2>
+          <table className="table table-bordered">
+            <tbody>
+              {missionprofile.map((missionprofile) => (
+                <tr key={missionprofile}>
+                  <th>{missionprofile}</th>
 
```

Comment:
- [OPTIONAL] The `My Profile` section can be enhanced by adding a placeholder message when the "My Missions" or "My Rockets" lists are empty (no missions joined or no rockets reserved).

- [OPTIONAL] The `My Profile` section can be enhanced by adding the "Cancel reservation" and "Leave Mission" buttons to the lists here. Clicking them should dispatch the actions you have already used in the main Rockets and Missions sections.

- [OPTIONAL] The `My Profile` section can be enhanced by adding the "Read more" button for each mission and rocket. Upon click, it should open a corresponding Wikipedia page in a new tab. NOTE - you need to get that extra Wikipedia URL from the API's payload.